src/main/java/com/palantir/docker/compose/DockerComposeRule.java

```diff
@@ -102,7 +121,20 @@ protected LogCollector logCollector() {
     public void before() throws IOException, InterruptedException {
         log.debug("Starting docker-compose cluster");
         dockerCompose().build();
-        dockerCompose().up();
+
+        try {
+            dockerCompose().up();
+        } catch (DockerExecutionException e) {
+            Set<String> conflictingContainerNames = getConflictingContainerNames(e.getMessage());
```

Comment:
This no longer seems like exceptional behaviour if we're able to handle and recover from it in hopefully the most common situations.

How about making dockerCompose().up() return an object containing the result? The parsing of the exception names could then move into this object which could be unit tested much more easily than having it all in this class.